src/PhpBrew/VariantBuilder.php

```diff
@@ -561,6 +562,39 @@ public function __construct()
             }
         };
 
+        /*
+         * --with-sodium option
+         *
+         * --with-sodium=shared
+         * --with-sodium=[dir]
+         *
+         * On ubuntu you need to install libsodium-dev
+         */
+        $this->variants['sodium'] = function (Build $build, $val = null) {
+            if ($val) {
+                return "--with-sodium=$val";
+            }
+
+            if ($prefix = Utils::findIncludePrefix('sodium.h')) {
```

Comment:
Could you please try using prefix finder like it's done for `libxml`?

https://github.com/phpbrew/phpbrew/blob/01de86c80079696899782c89922f8d69dbc0b9dd/src/PhpBrew/VariantBuilder.php#L760-L769

This way, the code is more expressive and is easier to maintain.